clients/rust/tests/harvest_holder_rewards.rs

```diff
@@ -0,0 +1,783 @@
+    account.lamports = account.lamports.saturating_add(4_000_000_000);
+    let rewards_lamports = account.lamports;
+    context.set_account(&config_manager.vault, &account.into());
+
+    // And a stake account wiht a 50 staked amount.
+
+    let stake_manager = StakeManager::new(&mut context, &config_manager.config).await;
+
+    let mut account = get_account!(context, stake_manager.stake);
+    let mut stake_account = Stake::from_bytes(account.data.as_ref()).unwrap();
+    // "manually" set the amount to 50
+    stake_account.amount = 50;
+
+    account.data = stake_account.try_to_vec().unwrap();
+    context.set_account(&stake_manager.stake, &account.into());
+
+    // And we initialize the holder rewards accounts.
+
+    let holder_rewards_pool = create_holder_rewards_pool(
+        &mut context,
+        &config_manager.mint,
+        &config_manager.mint_authority,
+    )
+    .await;
+
+    let holder_rewards = create_holder_rewards(
+        &mut context,
+        &holder_rewards_pool,
+        &config_manager.mint,
+        &config_manager.vault,
+    )
+    .await;
+
+    let mut account = get_account!(context, holder_rewards);
+    let mut holder_rewards_account = HolderRewards::from_bytes(account.data.as_ref()).unwrap();
+    // "manually" set last accumulated rewards per token to 40_000_000 (0.04 SOL)
+    holder_rewards_account.last_accumulated_rewards_per_token = 40_000_000 * 1_000_000_000;
+
+    account.data = holder_rewards_account.try_to_vec().unwrap();
+    context.set_account(&holder_rewards, &account.into());
+
+    // When we harvest the holder rewards.
+    //
+    // We are expecting the rewards to be 2 SOL.
+    //
+    // Calculation:
+    //   - total staked: 100
+    //   - holder rewards: 4 SOL
+    //   - rewards per token: 4_000_000_000 / 100 = 40_000_000 (0.04 SOL)
+    //   - rewards for 50 staked: 40_000_000 * 50 = 2_000_000_000 (2 SOL)
+
+    let destination = Pubkey::new_unique();
+
+    let harvest_holder_rewards_ix = HarvestHolderRewardsBuilder::new()
+        .config(config_manager.config)
+        .stake(stake_manager.stake)
+        .vault(config_manager.vault)
+        .holder_rewards(holder_rewards)
+        .vault_authority(find_vault_pda(&config_manager.config).0)
+        .stake_authority(stake_manager.authority.pubkey())
+        .destination(destination)
+        .mint(config_manager.mint)
+        .token_program(spl_token_2022::ID)
+        .instruction();
+
+    let tx = Transaction::new_signed_with_payer(
+        &[harvest_holder_rewards_ix],
+        Some(&context.payer.pubkey()),
+        &[&context.payer, &stake_manager.authority],
+        context.last_blockhash,
+    );
+    context.banks_client.process_transaction(tx).await.unwrap();
+
+    // Then the destination account has the rewards.
+
+    let account = get_account!(context, destination);
+    assert_eq!(account.lamports, 2_000_000_000);
+
+    // And there should be rewards left on the vault account.
```

Comment:
Maybe just out of an abundance of caution, check that the vault authority account didn't hang onto any lingering lamports?